src/compression_wrapper.c

```diff
@@ -171,91 +175,59 @@ cr_detect_compression(const char *filename, GError **err)
+               g_str_has_suffix(filename, ".txt"))
     {
         return CR_CW_NO_COMPRESSION;
     }
 
     // No success? Let's get hardcore... (Use magic bytes)
-
-    magic_t myt = magic_open(MAGIC_MIME | MAGIC_SYMLINK);
-    if (myt == NULL) {
-        g_set_error(err, ERR_DOMAIN, CRE_MAGIC,
-                    "magic_open() failed: Cannot allocate the magic cookie");
+    // Buffer to read the magic bytes into
+    char magic[5];
+    FILE* file = fopen(filename, "rb");
+    if (file == NULL) {
+        g_debug("%s: Unable to open file! (%s)", __func__, filename);
+        g_set_error(err, ERR_DOMAIN, CRE_IO, "fopen(): %s", g_strerror(errno));
         return CR_CW_UNKNOWN_COMPRESSION;
     }
 
-    if (magic_load(myt, NULL) == -1) {
-        g_set_error(err, ERR_DOMAIN, CRE_MAGIC,
-                    "magic_load() failed: %s", magic_error(myt));
-        return CR_CW_UNKNOWN_COMPRESSION;
+    size_t bytesRead = fread(magic, 1, sizeof(magic), file);
+    if (bytesRead != sizeof(magic)) {
+        // Assume that if there's less than 5 bytes in the file, it's uncompressed
+        g_debug("%s: Unable to read bytes from file for magic number detection, assuming uncompressed (%s)",
+            __func__, filename);
+        return CR_CW_NO_COMPRESSION;
     }
-
-    const char *mime_type = magic_file(myt, filename);
-
-    if (mime_type) {
-        g_debug("%s: Detected mime type: %s (%s)", __func__, mime_type,
-                filename);
-
-        if (g_str_has_prefix(mime_type, "application/x-gzip") ||
-            g_str_has_prefix(mime_type, "application/gzip") ||
-            g_str_has_prefix(mime_type, "application/gzip-compressed") ||
-            g_str_has_prefix(mime_type, "application/gzipped") ||
-            g_str_has_prefix(mime_type, "application/x-gzip-compressed") ||
-            g_str_has_prefix(mime_type, "application/x-compress") ||
-            g_str_has_prefix(mime_type, "application/x-gunzip") ||
-            g_str_has_prefix(mime_type, "multipart/x-gzip"))
-        {
-            type = CR_CW_GZ_COMPRESSION;
-        }
-
-        else if (g_str_has_prefix(mime_type, "application/zstd"))
-        {
-            type = CR_CW_ZSTD_COMPRESSION;
-        }
-
-        else if (g_str_has_prefix(mime_type, "application/x-bzip2") ||
-                 g_str_has_prefix(mime_type, "application/x-bz2") ||
-                 g_str_has_prefix(mime_type, "application/bzip2") ||
-                 g_str_has_prefix(mime_type, "application/bz2"))
-        {
-            type = CR_CW_BZ2_COMPRESSION;
-        }
-
-        else if (g_str_has_prefix(mime_type, "application/x-xz"))
-        {
-            type = CR_CW_XZ_COMPRESSION;
+    fclose(file);
+
+    // assume that most compressed files will have a high bit set somewhere in the first 5 bytes
+    // and most uncompressed files (generally text) will not
+    bool is_text_heuristic = true;
+    for (int i=0; i<bytesRead; i++) {
+        if (magic[i] > 127) {
```

Comment:
I believe this comparison is always `false`, since `magic` is a `char` array and `char` has a range of -128 to 127 .